src/ucs/datastruct/ptr_array.h

```diff
@@ -203,11 +203,12 @@ __ucs_ptr_array_for_each_get_step_size(ucs_ptr_array_t *ptr_array,
     uint32_t size_elem;
     ucs_ptr_array_elem_t elem = ptr_array->start[element_index];
 
-    if (ucs_unlikely(__ucs_ptr_array_is_free((ucs_ptr_array_elem_t)elem))) {
-       size_elem = (elem >> UCS_PTR_ARRAY_FREE_AHEAD_SHIFT);
-    } else {
+    size_elem = (elem >> UCS_PTR_ARRAY_FREE_AHEAD_SHIFT);
+    if (ucs_likely(!__ucs_ptr_array_is_free((ucs_ptr_array_elem_t)elem))) {
        size_elem = 1;
     }
+    /* Prefetch the next item */
+    ucs_prefetch((const void *)&(ptr_array)->start[element_index + size_elem]);
```

Comment:
1. excess ( ) around ptr_array
2. why need cast to const void*

Reply:
1. Will fix soon.
2. ucs_prefetch() is defined as follows,
src/ucs/sys/compiler_def.h:#define ucs_prefetch(p)            __builtin_prefetch(p)
Then, __builtin_prefetch() addr argument type is  (const void *)
__builtin_prefetch (const void * addr , ...).